Introduce Managed Identity configuration

Anders-Toegersen opened the pull request.

Commit:
Implemented Factories to use TokenCredential Authorization

Commit:
Extended ServiceCollectionExtensions configuration
Added tests

Commit:
Updated SampleApi and README

Noblix commented on Atc.Azure.Messaging.sln:
```diff
@@ -1,31 +1,31 @@
-	EndGlobalSection
-EndGlobal
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.0.31903.59
+MinimumVisualStudioVersion = 15.0.26124.0
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Atc.Azure.Messaging", "src\Atc.Azure.Messaging\Atc.Azure.Messaging.csproj", "{2F7702F2-A407-41FB-8C88-7C066237BC75}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Atc.Azure.Messaging.Tests", "test\Atc.Azure.Messaging.Tests\Atc.Azure.Messaging.Tests.csproj", "{0C49EE44-DAE3-4A7C-9D2D-D1190CAC85F0}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{2F7702F2-A407-41FB-8C88-7C066237BC75}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{2F7702F2-A407-41FB-8C88-7C066237BC75}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{2F7702F2-A407-41FB-8C88-7C066237BC75}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{2F7702F2-A407-41FB-8C88-7C066237BC75}.Release|Any CPU.Build.0 = Release|Any CPU
+		{0C49EE44-DAE3-4A7C-9D2D-D1190CAC85F0}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{0C49EE44-DAE3-4A7C-9D2D-D1190CAC85F0}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{0C49EE44-DAE3-4A7C-9D2D-D1190CAC85F0}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{0C49EE44-DAE3-4A7C-9D2D-D1190CAC85F0}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
+		SolutionGuid = {1EC093F7-1E03-4088-BA72-0DEB39AFEE82}
+	EndGlobalSection
+EndGlobal
```
It looks like nothing is changed here?

Anders-Toegersen replied:
That is correct.

Anders-Toegersen replied:
Not sure why it is shown as it is

christianhelle replied:
They are just whitespaces. Not sure why, but I'm guessing Rider writes a solution file differently than Visual Studio, or maybe it was created on a non-Windows machine... When you enable "hide whitespaces" the changes disappear...

Noblix commented on src/Atc.Azure.Messaging/EventHub/EventHubCredentialsPublisherFactory.cs:
```diff
@@ -0,0 +1,36 @@
+
+namespace Atc.Azure.Messaging.EventHub;
+
+#pragma warning disable CA2000 // Dispose objects before losing scope
```
I think this should be restored at the end of the file? Or use the SuppressMessage attribute instead?

christianhelle replied:
I prefer to use the `SuppressMessage` on the method level

Anders-Toegersen replied:
fixed

christianhelle commented on src/Atc.Azure.Messaging/EventHub/EventHubCredentialsPublisherFactory.cs:
```diff
@@ -0,0 +1,36 @@
+
+#pragma warning disable CA2000 // Dispose objects before losing scope
+
+public class EventHubCredentialsPublisherFactory : IEventHubPublisherFactory
```
Since this is an open source project and that we now have consumers of this project, I suggest we add XML code documentation comments to all public facing types

Just a simple `<summary>` should be enough

```
<summary>
Something something blah blah blah
</summary>
```

Anders-Toegersen replied:
Done

christianhelle commented on src/Atc.Azure.Messaging/ServiceBus/ServiceBusCredentialsClientFactory.cs:
```diff
@@ -0,0 +1,31 @@
+
+namespace Atc.Azure.Messaging.ServiceBus;
+
+public class ServiceBusCredentialsClientFactory : IServiceBusClientFactory
```
We can consider to be more strict and only expose interfaces, keeping implementation classes `internal`

Anders-Toegersen replied:
done

Commit:
Added SupressMessage with justification for CA2000

christianhelle commented on README.md:
````diff
@@ -37,6 +37,46 @@ builder.Services.AddSwaggerGen();
 builder.Services.ConfigureMessagingServices(builder.Configuration);
 ```
 
+## Connecting with Managed Identity
````
Brilliant!

christianhelle commented on sample/SampleApi/Program.cs:
```diff
@@ -12,7 +12,7 @@
 
 // Register Atc.Azure.Messaging dependencies
-builder.Services.ConfigureMessagingServices(builder.Configuration);
+builder.Services.ConfigureMessagingServices(builder.Configuration); // ConfigureMessagingServices(builder.Configuration, true)
```
Why is there a `//` at the end of line 15 ?

Anders-Toegersen replied:
fixed

perkops commented on src/Atc.Azure.Messaging/EventHub/EventHubCredentialsPublisherFactory.cs:
```diff
@@ -0,0 +1,39 @@
+using System.Diagnostics.CodeAnalysis;
```
Move to GlobalUsings

perkops commented on src/Atc.Azure.Messaging/ServiceBus/ServiceBusCredentialsClientFactory.cs:
```diff
@@ -0,0 +1,31 @@
+using Atc.Azure.Options.Authorization;
```
Move to GlobalUsings

perkops commented on src/Atc.Azure.Messaging/ServiceCollectionExtensions.cs:
```diff
@@ -1,26 +1,68 @@
-﻿using System.Diagnostics.CodeAnalysis;
```
Move to GlobalUsings

Commit:
Added XML comments to public interfaces
Changed accessors for classes
Added coding rules for uninstantiated classes
Exposed internal classes to Test assembly

perkops commented on test/Atc.Azure.Messaging.Tests/EventHub/EventHubCredentialsPublisherFactoryTests.cs:
```diff
@@ -0,0 +1,60 @@
+using Atc.Azure.Messaging.EventHub;
```
Move to GlobalUsings

perkops commented on test/Atc.Azure.Messaging.Tests/ServiceBus/ServiceBusCredentialsClientFactoryTests.cs:
```diff
@@ -0,0 +1,59 @@
+using Atc.Azure.Messaging.ServiceBus;
```
Move to GlobalUsings

perkops commented on test/Atc.Azure.Messaging.Tests/ServiceCollectionExtensionsTests.cs:
```diff
@@ -0,0 +1,73 @@
+using Atc.Azure.Messaging.EventHub;
```
Move to GlobalUsings
